docs/poet_operator_manual.md

```diff
@@ -0,0 +1,115 @@
+
+```.
+.
+├── layercache_0.bin
+├── layercache_10.bin
+├── layercache_1.bin
+├── layercache_2.bin
+├── layercache_3.bin
+├── layercache_4.bin
+├── layercache_5.bin
+├── layercache_6.bin
+├── layercache_7.bin
+├── layercache_8.bin
+├── layercache_9.bin
+└── state.bin
+```
+
+That directory listing is from a poet that was having round `3` in `in-progress` state, because as you can see there are layercache files. Each of these files is a layer (level) of the PoSW tree.
+
+Each of these files should be half of the size of the previous one (`0` being biggest).
+
+
+After round is finished poet will delete the tree from the disk and start a new round after `cycle_gap` ends.
+
+### Layercache
+
+PoET tries to optimize the cache usage by knowing how much `lps` (leafs per second) is it expected to get. You can specify that with `--lps` flag. To obtain the value the best is to use `poet_bench` tool but please run it for significant amount of time (at least 10 minutes) to get the stable results.
+
+By default poet uses 26 memory layers ( Up to (1 << 26) * 2 - 1 Merkle tree cache nodes (32 bytes each) will be held in-memory), the other layers are stored on disk. If your poet generates 10_000 leafs per second then the expected disk storage for a 13,5days running would be `10_000 * 60 * 60 * 24 * 13,5 = 11_664_000_000` leafs. That means that the produced tree will have `11_664_000_000`. But to produce `11_664_000_000` leafs you need `11_664_000_000 - 1` nodes above them. So we can multiply `11_664_000_000` by 2 to get the total tree size which is: `23_328_000_000`. If you multiply it by `32bytes` (size of one node) then you get `373.248 * 2 GB` which is `746,496 GB` for a PoET service that runs for 13,5 days and generates 10_000 leafs per second. That's an approximation because the tree is not always full. But it's a good enough approximation.
+
+Layers are written to a disk with buffered file, you can configure the buffer size by using `tree-file-buffer`, the default is `4096` and should be good enough for most cases.
+
+You can also use that logic to estimate needed disk performance and IOps. But HDD without a SSD cache should be more than sufficient in any case.
+
+You *need* to make sure that the data is persisted. The existence of the tree is essential to generate a proof for a given round. If the tree is not there then the proof will be impossible to properly generate.
+
+> [!NOTE]  
+> PoET will *not* start generating tree when it will detect that the round should be already in progress but the state of the round is empty. This is to prevent the situation when PoET is doing work for nothing
+
+## Performance fine tunning
+
+Because PoET is single threaded and CPU bound, it is important to fine tune the CPU and scheduler to get the best performance. On Linux platform we strongly recommend using kernel 6.2 and newer.
```

Comment:
Only the worker building the PoSW is single-threaded. The registration is multithreaded.

Reply:
Yes but I focused here about the PoSW, will add clarification.